compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/viewinterop/InteropPlatformDetails.skiko.kt

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2024 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.ui.viewinterop
+
+import androidx.compose.ui.Modifier
+
+/**
+ * An interface for platform-specific configurable details of how interop is implemented on a specific
+ * platform.
+ *
+ * It's needed in case the [InteropViewHolder] is reused with new properties and the modifier
+ * chain needs to be changed within ComposeNode Updater
+ */
+internal interface InteropPlatformDetails {
+    fun platformModifier(holder: InteropViewHolder): Modifier
```

Comment:
~Why isn't it a part of `InteropContainer`? Those 2 booleans don't seem as platform-specific and can be mentioned in the common interface as parameters in this function.~

or. it might be converted to something like

```kt
internal fun interface InteropViewModifierFactory {
    fun create(holder: InteropViewHolder): Modifier
}
```

and probably make some default implementation(s) in shared code (outside of this PR).

Reply:
Native platform resolution is absolutely platform specific, touch interaction mode is also platform specific.
The point of `InteropPlatformDetails` is an ability to change something in the implementation of specific `InteropViewHolder` when it's reused. It can cover basically anything what happens here (in case of iOS - not only change of a modifier, but also - reconfiguration of `InteropWrappingView` which serves as a `group`. In future it could be used for expanding gesture recogniser failure requirements behavior and forwarding it to the user.